### PR TITLE
Added check to see if delivery option is an array

### PR DESCRIPTION
**Pull request by nicosomb**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | 8.0.x
| Description?      | We tried to remove a key on a non array variable. 
| Type?             | bug fix 
| Category?         | FO 
| BC breaks?        | no
| Deprecations?     |  no
| Fixed ticket?     | Fixes https://github.com/PrestaShop/PrestaShop/issues/30078
| Related PRs       | 
| How to test?      | See https://github.com/PrestaShop/PrestaShop/issues/30078
| Possible impacts? |

### CODE REVIEW

**Review by mflasquin — DISMISSED** [review]

**Review by Hlavtox — DISMISSED** [review]

**Review by kpodemski — DISMISSED** [review]

**Review by matks — DISMISSED** [review]

**Review by PululuK — DISMISSED** [review]

**Review by jolelievre — DISMISSED** [review]
Hi @nicosomb 

seems like this one is not critical enough, it will be fixed in the patch so you should change your PR's target to 8.0.x, please. (validated by @MatShir )

**Comment by nicosomb** [comment]
arf I made mistake. @jolelievre I will fix that.

**Comment by nicosomb** [comment]
@jolelievre @MatShir done :tada:

**Comment by MhiriFaten** [comment]
Hi, 
This PR should be approved by @PrestaShop/prestashop-maintainers 
I will remove the 'waiting for QA' label.
Thanks!

**Review by Hlavtox — DISMISSED** [review]

**Review by lartist — DISMISSED** [review]

**Review by kpodemski — DISMISSED** [review]

File: classes/checkout/CheckoutPaymentStep.php [hunk 1]
```diff
@@ -84,7 +84,10 @@ public function render(array $extraParams = [])
         } else {
             $selectedDeliveryOption = 0;
         }
-        unset($selectedDeliveryOption['product_list']);
+
+        if (true === is_array($selectedDeliveryOption)) {
+            unset($selectedDeliveryOption['product_list']);
```
Comment (jolelievre) [inline classes/checkout/CheckoutPaymentStep.php:89]: Just a question, does it mean sometimes `$selectedDeliveryOption` is an array, sometimes it's not? Maybe that's an issue no? Seems like you're patching the symptom but the root cause remains, there is some inconsistency on this variable that is weird.

Starting with the previous line which sets the fallback value to 0, not sure it makes much sense.. Or maybe the unset should simply be moved into the `if (isset($deliveryOptions[$deliveryOptionKey])) {` block

Comment (nicosomb) [reply classes/checkout/CheckoutPaymentStep.php:89]: In the case I tested, `$selectedDeliveryOption` was null.

Comment (kpodemski) [reply classes/checkout/CheckoutPaymentStep.php:89]: @nicosomb, how is that? it's either value of `$deliveryOptions[$deliveryOptionKey]` or `0` 🤔 and there's `isset($deliveryOptions[$deliveryOptionKey]` above

+ there's a check to see if it's an array, but there's no check if the key exists, maybe something like:

`is_array($selectedDeliveryOption) && isset($selectedDeliveryOption['product_list'])` would be more precise?

**Review by kpodemski — APPROVED** [review]

**Review by matks — APPROVED** [review]

**Review by sallemiines — COMMENTED** [review]
Hello @nicosomb !
Thanx for the PR, it's LGTM QA :heavy_check_mark: !

80xpr

Thank you !

**Comment by jolelievre** [comment]
Thanks @nicosomb and @sallemiines

**Comment by MatShir** [comment]
🎉

**Pull request merged**